common/soroban-rpc/bin/start

```diff
@@ -3,14 +3,6 @@
 set -e
 set -o pipefail
 
- while ! $(curl --silent --location --request GET 'http://localhost:8001' | \
-            jq --exit-status '.core_latest_ledger > 5 and .history_latest_ledger > 5' 2>/dev/null | \
-            grep -o true || echo false );
- do
-    echo "Waiting for horizon to be available..."
-    sleep 1
-done
-
```

Comment:
@sreuland now that we are removing Horizon, should we wait for the checkpoint to be reached? (soroban-rpc is not usable until then)

I am considering to make the soroban-rpc health endpoint access the DB to only mark it as healthy once the DB is responding (i.e. initialized and the checkpoint reached)

Reply:
Related: https://github.com/stellar/soroban-tools/pull/449

Reply:
this could wait for checkpoint pub'd in archive url first, as that is dependency of soroban rpc ingestion, but, I think unnecessary, this can proceed with removal of wait loop here, but would work best once [stellar/soroban-tools#449](https://github.com/stellar/soroban-tools/pull/449) is merged first. 

supervisord config in quickstart manages soroban rpc service and restarts it on an increasing backoff interval, so, by default provides resiliency for rpc service to eventually come up in ready state once a checkpoint has been published, and any wait attempted here is redundant attempt at optimizing which layer does the wait at best.

there's a related part in quickstart main `start` script - https://github.com/stellar/quickstart/blob/master/start#L565, it displays readiness status of soroban rpc on console logs, I think we want to change that for `ENABLE_SOROBAN_RPC =true` path to not wait on horizon status first- https://github.com/stellar/quickstart/blob/master/start#L583

once [stellar/soroban-tools#449](https://github.com/stellar/soroban-tools/pull/449) merges, then the check on soroban rpc health endpoint for readiness will become much more effective - https://github.com/stellar/quickstart/blob/master/start#L589. supervisord will still need to re-start rpc until the checkpoint happens, that may quiet down further once other related ticket comes in https://github.com/stellar/soroban-tools/issues/358

Reply:
OK, I've changed the `service_status()` so that `horizon_status` runs in the backgound. PTAL